ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp

```diff
@@ -165,17 +167,23 @@ wid_t window_get_identifier(window_t hwnd) {
   return std::to_string(reinterpret_cast<unsigned long long>(hwnd));
 }
 
+static int currentIconIndex = -1;
+static unsigned currentIconFrame;
+
 void window_set_visible(bool visible) {
   if (visible) {
     XMapRaised(disp, win);
+    if (!sprite_exists(currentIconIndex)) {
+      XSynchronize(disp, True);
+      Atom property = XInternAtom(disp, "_NET_WM_ICON", False);
+      XChangeProperty(disp, win, property, XA_CARDINAL, 32, PropModeReplace, (unsigned char *)xwindow_icon_default, 3);
+      XFlush(disp);
+    }
```

Comment:
This does not go here, move it to where the window is created.

Reply:
Could you please explain a better place to put it that doesn't segfault the game like putting it where the window is being created currently does without a major revamp of the entire system?

Reply:
The reason is segfaults is normal xlib behavior because the icon can not be set unless the window is visible. As I mentioned on discord another place i could put it is outside the xlib platform entirely using a #if defined(X_PROTOCOL) macro where the window first is set to visible which I know you also won't like.